Source/WTF/wtf/text/CStringView.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2018 Yusuke Suzuki <utatane.tea@gmail.com>
+ * Copyright (C) 2024 Apple Inc. All Rights Reserved.
+ * Copyright (C) 2025 Comcast Inc.
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <span>
+#include <string>
+#include <type_traits>
+#include <wtf/Compiler.h>
+#include <wtf/Forward.h>
+#include <wtf/HashFunctions.h>
+#include <wtf/StdLibExtras.h>
+#include <wtf/text/ASCIILiteral.h>
+#include <wtf/text/SuperFastHash.h>
+
+namespace WTF {
+
+class PrintStream;
+class String;
+
+class CStringView final {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    operator const char*() const LIFETIME_BOUND { return m_charactersWithNullTerminator.data(); }
+
+    static CStringView fromNullTerminatedStringUnsafe(const char* string LIFETIME_BOUND)
```

Comment:
It's an anti-pattern in `CString` that you construct a `CString`, and get a buffer pointer from a `CString`, without specifying an encoding.

Instead, we want `CString` to behave like `String`, and operate on `span<LChar>` (Latin1 encoding), `span<UChar>` (UTF-16 encoding), or use an explicit conversion function like `fromLatin1(char*)`.

It's going to take some time to fix up CString and all its clients, but we can at least avoid expanding the unspecified encoding mistake in new code.

So I think the `CStringView` class needs an `m_is8Bit` member like `StringView` has, and I think it needs `span8` and `span16` accessors that ASSERT (rather than `operator char*`), and constructors that know what they're constructing.